en/simulation/ros_interface.md

```diff
@@ -1,50 +1,53 @@
 
-If an interface to ROS is wanted, the already running secondary MAVLink instance can be connected to ROS via [mavros](../ros/mavros_offboard.md). To connect to a specific IP (`fcu_url` is the IP / port of SITL), use a URL in this form:
+![PX4 SITL overview](../../assets/simulation/px4_sitl_overview.png)
 
-<div class="host-code"></div>
+The ROS/Gazebo integration with PX4 follows the pattern above. The only *slight* difference being that ROS initiates the connection on port 14557, while it is more typical for an offboard API to listen for connections on UDP port 14540. 
 
-```sh
-roslaunch mavros px4.launch fcu_url:="udp://:14540@192.168.1.36:14557"
-```
 
-To connect to localhost, use this URL:
+## Installing Gazebo for ROS
 
-<div class="host-code"></div>
+The easiest way to setup PX4 simulation with ROS on Ubuntu Linux is to use the standard installation script that can be found at [Development Environment on Linux > Gazebo with ROS](../setup/dev_env_linux.md#gazebo-with-ros). The script installs everything you need: PX4, ROS "Kinetic", the Gazebo 7 simulator, and [MAVROS](../ros/mavros_installation.md).
 
-```sh
-roslaunch mavros px4.launch fcu_url:="udp://:14540@127.0.0.1:14557"
-```
+> **Note** The script follows the [standard ROS "Kinetic" installation instructions](http://wiki.ros.org/kinetic/Installation/Ubuntu), which include Gazebo 7. Installation of ROS Kinetic for other platforms is covered in the [ROS Wiki here](http://wiki.ros.org/kinetic/Installation).
 
 
-## Installing Gazebo for ROS
+## Launching ROS/Simulation
 
-The Gazebo ROS SITL simulation is known to work with both Gazebo 6 and Gazebo 7, which can be installed via:
+The command below can be used to launch the simulation and connect ROS to it via [MAVROS](../ros/mavros_installation.md), where `fcu_url` is the IP / port of the computer running the simulation:
 
 ```sh
-sudo apt-get install ros-$(ROS_DISTRO)-gazebo7-ros-pkgs    //Recommended
+roslaunch mavros px4.launch fcu_url:="udp://:14540@192.168.1.36:14557"
 ```
-or
+
+To connect to localhost, use this URL:
+
 ```sh
-sudo apt-get install ros-$(ROS_DISTRO)-gazebo6-ros-pkgs
+roslaunch mavros px4.launch fcu_url:="udp://:14540@127.0.0.1:14557"
 ```
 
-## Launching Gazebo with ROS wrappers
+> **Tip** It can be useful to call *roslaunch* with the `-w` (warn) and/or `-v` (verbose) in order to get warnings about missing dependencies in your setup. For example:
+  ```sh
+  roslaunch mavros px4.launch fcu_url:="udp://:14540@127.0.0.1:14557"
+  ```
+
 
-In case you would like to modify the Gazebo simulation to integrate sensors publishing directly to ROS topics e.g. the Gazebo ROS laser plugin, Gazebo must be launched with the appropriate ROS wrappers.
+## Launching Gazebo with ROS Wrappers
```

Comment:
@TSC21 So how I interpret this is that rather than having ROS get rangefinder information from PX4 it can get it from gazebo. 
- Does this work with PX4 currently?
- If so, how does the information get to ROS? - we currently have no "path" in our comms directly between ROS and gazebo - they both just talk to PX4. 
- If so, where/when is this useful?

Reply:
The PX4 ROS implementation (which is an active part of SITL) has everything built in to receive and publish data in the ROS framework - but I actually don't know what is happening there cause there seems to be some data handled as ROS msgs, while there is data also being handled as Mavlink msgs. @bkueng maybe you can clarify that?
But we also should consider to implement some time of interface with MAVROS (`rotors_simulator` has a specific plugin to parse all the Gazebo sim data to ROS topics, but we don't do that - we use gazebo msgs only). All the data that's sent to an autopilot (being it SITL or other) is parsed first or `gazebo_mavlink_interface` or on other plugins and published as gazebo msgs (that can be received on the PX4 ROS implementation).

Reply:
> @bkueng maybe you can clarify that?

Sorry, since I don't use ROS I don't know the internals on that.

Reply:
Ok but then whoever wrote the PX4 ROS implementation can give a hand here?

Reply:
This all sounds very inconsistent. It would be useful to understand, but I doubt we'll get a timely answer on this point. OK if I create a separate issue that this section needs update and merge the rest?

Reply:
What does require a clarification is how the PX4 ROS implementation is done, what it accepts and how it can be applied generally (if it can be pulled out of the SITL sim environment for example).

Reply:
Yes. I have posted a question https://github.com/PX4/Devguide/issues/399 for that.